Fix app-defaults install path.
Fix link to libxaw3dxft path.
Fix race condition (See also: https://lists.freebsd.org/archives/freebsd-ports/2023-July/004114.html)

--- Makefile.am.orig	2021-01-07 14:18:14 UTC
+++ Makefile.am
@@ -3,7 +3,7 @@ SHAREDIR = $(prefix)/share/xpaint
 ## Local Configuration arguments
 BINDIR = $(prefix)/bin
 SHAREDIR = $(prefix)/share/xpaint
-XAPPLOADDIR = /etc/X11/app-defaults
+XAPPLOADDIR = $(prefix)/share/X11/app-defaults
 
 ## Enable/disable PRINTCAP (for older BSD style printing configuration)
 ## PRINTCAP = -DPRINTCAP
@@ -85,7 +85,7 @@ version.h:
 .PHONY:
 
 version.h:
-	rm -f xaw_incdir ; ln -s /usr/include/X11/Xaw3dxft xaw_incdir
+	rm -f xaw_incdir ; ln -s ${LOCALBASE}/include/X11/Xaw3dxft xaw_incdir
 	echo "#define XPAINT_VERSION \"$(PACKAGE_VERSION)\"" > version.h
 
 substads: substads.c
@@ -100,7 +100,7 @@ messages.h: preproc
 messages.h: preproc
 	./preproc > messages.h
 
-XPaint.ad: app-defaults
+XPaint.ad: substads app-defaults
 	$(CC) $(CFLAGS) substads.c -o substads
 	cd app-defaults ; \
 	../substads -appdefs \
